app/src/main/java/agency/five/codebase/android/movieapp/ui/component/MovieCategoryLabel.kt

```diff
@@ -0,0 +1,73 @@
+package agency.five.codebase.android.movieapp.ui.component
+
+import agency.five.codebase.android.movieapp.R
+import agency.five.codebase.android.movieapp.ui.theme.Blue
+import agency.five.codebase.android.movieapp.ui.theme.Gray300
+import agency.five.codebase.android.movieapp.ui.theme.Gray600
+import agency.five.codebase.android.movieapp.ui.theme.Gray700
+import androidx.annotation.StringRes
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.material.Text
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.Font
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.text.style.TextDecoration
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+
+
+
+sealed class MovieCategoryLabelTextViewState()
+data class MovieCategoryStringParam(val value: String) : MovieCategoryLabelTextViewState()
+data class MovieCategoryStringResource(@StringRes val value: Int) : MovieCategoryLabelTextViewState()
+
+data class MovieCategoryLabelViewState(
+    val itemId: Int,
+    val isSelected: Boolean,
+    val categoryText: MovieCategoryLabelTextViewState
+)
+
+@Composable
+fun MovieCategoryLabel(
+    movieCategoryLabelViewState: MovieCategoryLabelViewState,
+    modifier: Modifier =Modifier
+){
+    var isSelected by remember{
+        mutableStateOf(movieCategoryLabelViewState.isSelected)
+    }
+    Column(modifier = modifier.wrapContentWidth()   //width(IntrinsicSize.Max) ??
```

Comment:
why is it wrapped in column if you have only 1 view? I you wrapped it but forgot to uncomment box which represents indicator?

Reply:
Yes I wrapped it in column cause I wanted to use the box as an indicator but I don't know how to match the box width when the text will be different widths for different categories. I ended up using the TextDecoration.Underline as the indicator but it doesn't look as good as the box would. I left it commented in case I find how to match the box width to text width.